homeassistant/components/southern_company/coordinator.py

```diff
@@ -0,0 +1,181 @@
+                    str, southern_company_api.account.MonthlyUsage
+                ] = {}
+                for account in await self._southern_company_connection.accounts:
+                    _LOGGER.debug("Updating sensor data for %s", account.number)
+                    account_month_data[account.number] = await account.get_month_data(
+                        await self._southern_company_connection.jwt
+                    )
+                # Note: insert statistics can be somewhat slow on first setup.
+                await self._insert_statistics()
+                return account_month_data
+        except SouthernCompanyException as ex:
+            raise UpdateFailed("Failed updating jwt token") from ex
+
+        raise UpdateFailed("No jwt token")
+
+    async def _insert_statistics(self) -> None:
+        """Insert Southern Company statistics."""
+        if await self._southern_company_connection.jwt is None:
+            raise UpdateFailed("Jwt is None")
+        for account in await self._southern_company_connection.accounts:
+            _LOGGER.debug("Updating Statistics for %s", account.number)
+            cost_statistic_id = f"{DOMAIN}:energy_" f"cost_" f"{account.number}"
+            usage_statistic_id = f"{DOMAIN}:energy_" f"usage_" f"{account.number}"
+
+            last_stats = await get_instance(self.hass).async_add_executor_job(
+                get_last_statistics, self.hass, 1, usage_statistic_id, True, {}
+            )
+            if not last_stats:
+                # First time we insert 1 year of data (if available)
+                _LOGGER.info(
+                    "Updating statistic for the first time, this may take a while"
+                )
+                hourly_data = await account.get_hourly_data(
+                    datetime.datetime.now() - timedelta(days=365),
+                    datetime.datetime.now(),
+                    await self._southern_company_connection.jwt,
+                )
+
+                _cost_sum = 0.0
+                _usage_sum = 0.0
+                last_stats_time = None
+            else:
+                # hourly_consumption/production_data contains the last 30 days
+                # of consumption/production data.
+                # We update the statistics with the last 30 days
+                # of data to handle corrections in the data.
+                hourly_data = await account.get_hourly_data(
+                    datetime.datetime.now() - timedelta(days=31),
+                    datetime.datetime.now(),
+                    await self._southern_company_connection.jwt,
+                )
+
+                from_time = hourly_data[0].time
+                start = from_time - timedelta(hours=1)
+                cost_stat = await get_instance(self.hass).async_add_executor_job(
+                    statistics_during_period,
+                    self.hass,
+                    start,
+                    None,
+                    [cost_statistic_id],
```

Comment:
You can combine the two statistics_during_period calls by specifying [cost_statistic_id,usage_statistic_id] here